test/testrunner.py

```diff
@@ -121,19 +121,21 @@ def pull_dlc_images(images):
 
 
 def setup_eks_clusters(dlc_images):
-    terminable_clusters = []
     frameworks = {"tensorflow": "tf", "pytorch": "pt", "mxnet": "mx"}
-    for long_name, short_name in frameworks.items():
-        if long_name in dlc_images:
-            cluster_name = None
-            if not is_pr_context():
-                num_nodes = 3 if long_name != "pytorch" else 4
-                cluster_name = f"dlc-{short_name}-cluster-" \
-                               f"{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}-{random.randint(1, 10000)}"
-                eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
-                terminable_clusters.append(cluster_name)
-            eks_utils.eks_setup(long_name, cluster_name)
-    return terminable_clusters
+    frameworks_in_images = [framework for framework in frameworks.keys() if framework in dlc_images]
+    if len(frameworks_in_images) != 1:
+        raise ValueError(
+            f"All images in dlc_images must be of a single framework for EKS tests.\n"
+            f"Instead seeing {frameworks_in_images} frameworks."
+        )
+    long_name = frameworks_in_images[0]
+    short_name = frameworks[long_name]
+    num_nodes = 2 if is_pr_context() else 3 if long_name != "pytorch" else 4
+    cluster_name = f"dlc-{short_name}-cluster-" \
+                   f"{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}-{random.randint(1, 10000)}"
+    eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
+    eks_utils.eks_setup(long_name, cluster_name)
```

Comment:
There is a condition i just realized where create could succeed but setup could fail, and we would not properly clean up the cluster. I think it might be wise to wrap the eks setup call in try/finally and always return cluster name after we've created the eks cluster, so that in this case we won't leave clusters spun up